framework/mesh/mesh_generator/split_file_mesh_generator.cc

```diff
@@ -316,12 +317,13 @@ SplitFileMeshGenerator::ReadSplitMesh()
                          std::to_string(opensn::mpi_comm.size()) + " processes.");
 
   info_block.dimension = ReadBinaryValue<unsigned int>(ifile);
-  info_block.mesh_attributes_ = ReadBinaryValue<int>(ifile);
+  info_block.mesh_type = static_cast<MeshType>(ReadBinaryValue<int>(ifile));
```

Comment:
just curious what the static cast does here. it wasn't there before. what is an mistake?

Reply:
Originally `mesh_attributes_` were declared as `int` the `SplitMeshInfo` and the static cast was done `SplitFileMeshGenerator::SetupLocalMesh `. I changed the declaration of `SplitMeshInfo::mesh_attributes_` to be `MeshAttributes`, so the static cast was moved inside the `SplitFileMeshGenerator::ReadSplitMesh `.

Then, `mesh_attributes_` became `mesh_type`...